src/Entity/BannerSelection/Campaign.php

```diff
@@ -62,7 +60,7 @@ public function selectBucket( ?string $bucketId ): Bucket {
 		return $this->buckets[$this->rng->getRandomInteger( 0, count( $this->buckets ) - 1 )];
 	}
 
-	public function isRatioLimited(): bool {
-		return ( $this->rng->getRandomInteger( 1, 100 ) / 100 ) > $this->displayRatio;
+	public function getDisplayPercentage(): int {
```

Comment:
Why did you convert the more "active" method into a "dumb" getter, moving to logic to the use case? 

Just curious because I have a newly-developed aversion against getters, because many times they violate the "[Tell, don't ask](https://martinfowler.com/bliki/TellDontAsk.html)" principle. In this case it's not very clear-cut (`isRatioLimited` or a function with a better name is kind of getter-ish), but keeping the decision logic here would help avoid the use case depend on the RNG interface.

Reply:
I talked to @timEulitz and we realized that putting the logic in the use case makes more sense. @tzhelyazkova originally proposed this in https://github.com/wmde/banner-server/pull/19#pullrequestreview-169223065,